plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/utils/NsxControllerUtils.java

```diff
@@ -124,6 +124,9 @@ public static String getServerPoolName(String tier1GatewayName, long lbId) {
     }
 
     public static String getActiveMonitorProfileName(String lbServerPoolName, String port, String protocol) {
+        if (protocol.equalsIgnoreCase("udp")) {
+            protocol =  "ICMP";
+        }
```

Comment:
why is udp forbidden and why is it replaced by icmp? (next icmp + port is returned which is not realy a thing)

Reply:
This was a workaround to an issue found while enabling maintenance on an NSX edge node, after disabling maintenance the UDP load balancer wouldn't come up again as the active monitor failed instantiating, so the workaround was creating ICMP active monitors for UDP load balancers